### PR TITLE
add update product type

### PR DESCRIPTION
**Pull request by arnesuscharles**

### CODE REVIEW

File: proto/store-error.proto [hunk 1]
```diff
@@ -21,4 +21,5 @@ enum StoreErrorCode {
 	NAME_IS_REQUIRED = 17;
 	PRICE_IS_REQUIRED = 18;
 	ERROR_WHEN_GETTING_CLAIMS_FROM_JWT_TOKEN = 19;
+	ERROR_WHEN_INSERTING_OR_UPDATING_PRODUCT_TYPE = 20;
```
Comment (nurcholisnanda) [inline proto/store-error.proto]: based on our last discussion, I think we are starting to add errors in service/constant.go and following the format, so might need to check on this or delete this one if unnecessary

File: service/service.go [hunk 2]
```diff
@@ -442,7 +442,16 @@ func (s *service) UpdateProductCategory(ctx context.Context, prodCategory *prodE
 	return s.productRepository.UpsertProductCategory(ctx, prodCategory)
 }
 
-func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.ProductType) error {
+func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.ProductType, isUpdate bool) error {
+	if isUpdate {
+		if _, err := s.productRepository.GetProductTypeById(ctx, prodType.ID); err != nil {
```
Comment (nurcholisnanda) [inline service/service.go:448]: I think the current addition is already fine, but we need to change some logic on the previous code, such as GetProductTypeByName needs to return an error if the record is not found, so conditions like existingProdType can be deleted.

File: service/service.go [hunk 3]
```diff
@@ -442,7 +442,16 @@ func (s *service) UpdateProductCategory(ctx context.Context, prodCategory *prodE
 	return s.productRepository.UpsertProductCategory(ctx, prodCategory)
 }
 
-func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.ProductType) error {
+func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.ProductType, isUpdate bool) error {
```
Comment (paulusherdiyatma) [inline service/service.go]: why is the parameter named with isUpdate? could we change to another name that is related to the condition?

Comment (arnesuscharles) [reply service/service.go]: in grpc handler, both insert and update functions called this UpsertProductType function in the service package, so isUpdate parameter is used to differentiate whether the request is to update the product type (isUpdate = true) or to insert product type (isUpdate = false).
There will be an additional validation if the request is an update to the product type.
Do you have any suggestions for the parameter name mas?

Comment (paulusherdiyatma) [reply service/service.go]: can we check from the ID? I mean when we are inserting data, the ID is not provided and when updating data, the ID is provided from the FE. WDYT mas?

Comment (arnesuscharles) [reply service/service.go]: Yes I agreed and we can actually do that also, okay will try to apply that one, thank you mas

Comment (arnesuscharles) [reply service/service.go]: Updated ya mas

File: proto/store-error.proto (hunk 1 above)
Comment (arnesuscharles) [reply proto/store-error.proto]: updated mas, turned out this error in proto is not used

Comment (nurcholisnanda) [reply proto/store-error.proto]: thanks mas, LGTM from me

**Review by nurcholisnanda — APPROVED** [review]

**Pull request merged**